qi-lib/flow/core/compiler/deforest/cps.rkt

```diff
@@ -24,6 +25,13 @@
     [(_ [op (f ...) g ...] rest ...) (op f ... (inline-compose1 rest ...) g ...)]
     ))
 
+;; Adds the initial states of all stateful transformers in the
+;; required order to the initial producer state. Uses (cons Tx S)
+;; where Tx is the transformer's initial state and S is the producer's
+;; initial state with all preceding transformer states already
+;; added. Nothing is added for stateless transformers which pass () as
+;; their initial state expression. For example: (inline-consing (T1)
+;; () (T2) P) -> (cons T2 (cons T1 P))
```

Comment:
I think the `consing` parameter in later functions is used like a procedure, so
I'm confused. This change should also be a part of 1a79dc0 (Implement proper
state cons-ing for take., 2024-05-31).